File: articles/probability-for-systems-modeling/go/monte_carlo_mean.go
package main

import (
	"fmt"
	"math/rand"
)

func main() {
	const n = 10000
	rng := rand.New(rand.NewSource(42))

	total := 0.0

	for i := 0; i < n; i++ {
		exposure := 0.2 + 0.8*rng.Float64()
		vulnerability := rng.Float64()
		loss := exposure * vulnerability
		total += loss
	}

	fmt.Printf("Monte Carlo mean loss estimate: %.8f\n", total/float64(n))
}

File: articles/probability-for-systems-modeling/c/monte_carlo_mean.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    const int n = 10000;
    unsigned int seed = 42;
    double total = 0.0;

    srand(seed);

    for (int i = 0; i < n; i++) {
        double exposure = 0.2 + 0.8 * ((double) rand() / RAND_MAX);
        double vulnerability = (double) rand() / RAND_MAX;
        double loss = exposure * vulnerability;
        total += loss;
    }

    printf("Monte Carlo mean loss estimate: %.8f\n", total / n);

    return 0;
}
